### PR TITLE
fix: add pointer to `Closed` in PortalData and PreparedStatementData to avoid double close.

### PR DESCRIPTION
**Pull request by TianyuZhang1214**

### CODE REVIEW

File: pgserver/connection_handler.go [hunk 1]
```diff
@@ -922,15 +925,21 @@ func (h *ConnectionHandler) deletePreparedStatement(name string) {
 	ps, ok := h.preparedStatements[name]
 	if ok {
 		delete(h.preparedStatements, name)
-		ps.Stmt.Close()
+		if !ps.Closed.Load() {
```
Comment (fanyang01) [inline pgserver/connection_handler.go:928]: https://pkg.go.dev/sync/atomic#Bool.CompareAndSwap

File: pgserver/connection_handler.go [hunk 2]
```diff
@@ -922,15 +925,21 @@ func (h *ConnectionHandler) deletePreparedStatement(name string) {
...
+		}
 	}
 }
 
 func (h *ConnectionHandler) deletePortal(name string) {
 	p, ok := h.portals[name]
 	if ok {
 		delete(h.portals, name)
-		p.Stmt.Close()
+		if !p.Closed.Load() {
```
Comment (fanyang01) [inline pgserver/connection_handler.go:939]: https://pkg.go.dev/sync/atomic#Bool.CompareAndSwap

Comment (TianyuZhang1214) [reply pgserver/connection_handler.go]: fixed

File: pgserver/connection_handler.go (hunk 1 above)
Comment (TianyuZhang1214) [reply pgserver/connection_handler.go]: fixed

**Comment by TianyuZhang1214** [comment]
@fanyang01 
Maybe it's ready to merge now?

**Review by fanyang01 — APPROVED** [review]
LGTM

**Pull request merged**